package/MDAnalysis/coordinates/base.py

```diff
@@ -1014,6 +1021,63 @@ def add_auxiliary(self, auxname, auxdata, format=None, **kwargs):
         Auxiliary data is assumed to be time-ordered, with no duplicates. See
         the :ref:`Auxiliary API`.
         """
+        # auxdata can be an AuxReader or the data itself, try to get AuxReader
+        # from auxdata
+        try:
+            auxreader = get_auxreader_for(auxdata)
+        except AttributeError as e:
```

Comment:
Can we not have `_get_auxreader_for` give back a more intelligible error message than `NoneType' object has no attribute 'upper`?